pipeline_dp/budget_accounting.py

```diff
@@ -128,23 +134,35 @@ def request_budget(
     def compute_budgets(self):
         pass
 
-    def scope(self, weight: float):
-        """Defines a scope for DP operations that should consume no more than "weight" proportion of the budget
-        of the parent scope.
+    def scope(self,
+              weight: float,
+              aggregation_scope=False) -> 'BudgetAccountantScope':
+        """Returns a scope for DP operations
 
-        The accountant will automatically scale the budgets of all sub-operations accordingly.
+        The returned scope should consume no more than "weight" proportion of
+        the budget of the parent scope.
+
+        The accountant will automatically scale the budgets of all
+        sub-operations accordingly.
 
         Example usage:
           with accountant.scope(weight = 0.5):
              ... some code that consumes DP budget ...
 
         Args:
-            weight: budget weight of all operations made within this scope as compared to.
+            weight: budget weight of all operations made within this scope as
+             compared to.
+            aggregation_scope: if True, this is a high-level scope which
+             corresponds to one aggregation.
 
         Returns:
-            the scope that should be used in a "with" block enclosing the operations consuming the budget.
+            the scope that should be used in a "with" block enclosing the
+            operations consuming the budget.
         """
-        return BudgetAccountantScope(self, weight)
+        aggregation_index = None
```

Comment:
I am not sure if I perfectly understand this aggregation_scope parameter. Why do we have separate logic when it is set and when it is not?

Reply:
There were no yet comments about aggregation_scope, please check now

Reply:
Looks much better, thanks!